executor/executor.go

```diff
@@ -393,6 +394,28 @@ func RunTeardownOnAbort(path string, env []string, emitter screwdriver.Emitter,
 		return fmt.Errorf("Cannot start shell: %v", err)
 	}
 
+	// Command to Export Env. Use tmpfile just in case export -p takes some time
+	exportEnvCmd :=
+		"tmpfile=" + tmpFile + "; exportfile=" + exportFile + "; " +
+			"export -p | grep -vi \"PS1=\" > $tmpfile && mv $tmpfile $exportfile; "
+
+	// Run setup commands
+	setupCommands := []string{
+		"set -e",
+		"export PATH=$PATH:/opt/sd",
+		// trap EXIT, echo the last step ID and write ENV to /tmp/buildEnv
+		"finish() { " +
+			"EXITCODE=$?; " +
+			exportEnvCmd +
+			"echo $SD_STEP_ID $EXITCODE; }", //mv newfile to file
+		"trap finish EXIT;\necho ;\n",
+	}
+
+	setupReader := bufio.NewReader(f)
```

Comment:
will this work when abort happens ?

by the time this function runs https://github.com/screwdriver-cd/launcher/blob/cd4a62a4e0e156fc15576ac21989ce5eb01ab12b/launch.go#L667 , is the current running step already aborted ?

Reply:
yes, teardown is triggered via the trap handler

Reply:
what happens if they create/update the environment variable and want to use the updated value in the teardown step? it will not be possible and might need to revisit later.

Reply:
Looking at this build https://beta.cd.screwdriver.cd/pipelines/6512/builds/118185/steps/teardown-step1 
This PR still hasn't fixed the issue, right ? `ENV_VAR_TEST` should have been available ? 

I think need to have a different logic for env export which complements `trap`. 
At end of each step run, export the env data to a separate file and use this file to restore `env`

Reply:
seems this is an existing behavior env var exported in one of the steps is not available in the other.
https://beta.cd.screwdriver.cd/pipelines/6512/builds/118224/steps/deploy. It should be set via the environment configuration in sd.yaml